internal/indexmanagement/reconcile.go

```diff
@@ -140,65 +134,47 @@ func ReconcileCurationConfigmap(apiclient client.Client, cluster *apis.Elasticse
-	if err != nil {
-		return kverrors.Wrap(err, "failed to reconcile rollover cronjob", "policymapping", mapping.Name)
-	}
-	conditions := calculateConditions(policy, primaryShards)
-	name := fmt.Sprintf("%s-rollover-%s", cluster.Name, mapping.Name)
-	payload, err := json.Marshal(map[string]rolloverConditions{"conditions": conditions})
-	if err != nil {
-		return kverrors.Wrap(err, "failed to serialize the rollover conditions to JSON")
-	}
-	envvars := []corev1.EnvVar{
-		{Name: "PAYLOAD", Value: base64.StdEncoding.EncodeToString(payload)},
-		{Name: "POLICY_MAPPING", Value: mapping.Name},
-	}
-	fnContainerHandler := func(container *corev1.Container) {
-		container.Command = []string{"bash"}
-		container.Args = []string{
-			"-c",
-			"/tmp/scripts/rollover",
+		minAgeMillis, err := calculateMillisForTimeUnit(policy.Phases.Delete.MinAge)
+		if err != nil {
+			return err
 		}
+		envvars := []corev1.EnvVar{
+			{Name: "POLICY_MAPPING", Value: mapping.Name},
+			{Name: "MIN_AGE", Value: strconv.FormatUint(minAgeMillis, 10)},
+		}
+		containers = append(containers, newInitContainer(cluster.Name, "delete", constants.PackagedElasticsearchImage(), "/tmp/scripts/delete", envvars))
+	} else {
+		log.V(1).Info("Skipping curation cronjob for policymapping; delete phase not defined", "policymapping", mapping.Name)
 	}
-	desired := newCronJob(cluster.Name, constants.PackagedElasticsearchImage(), cluster.Namespace, name, schedule, cluster.Spec.Spec.NodeSelector, cluster.Spec.Spec.Tolerations, envvars, fnContainerHandler)
 
-	cluster.AddOwnerRefTo(desired)
-	return reconcileCronJob(apiclient, cluster, desired, areCronJobsSame)
-}
-
-func ReconcileCurationCronjob(apiclient client.Client, cluster *apis.Elasticsearch, policy apis.IndexManagementPolicySpec, mapping apis.IndexManagementPolicyMappingSpec, primaryShards int32) error {
-	if policy.Phases.Delete == nil {
-		log.Info("Skipping curation cronjob for policymapping; delete phase not defined", "policymapping", mapping.Name)
-		return nil
+	if policy.Phases.Hot != nil {
+		conditions := calculateConditions(policy, primaryShards)
+		payload, err := json.Marshal(map[string]rolloverConditions{"conditions": conditions})
+		if err != nil {
+			return kverrors.Wrap(err, "failed to serialize the rollover conditions to JSON")
+		}
+		envvars := []corev1.EnvVar{
+			{Name: "PAYLOAD", Value: base64.StdEncoding.EncodeToString(payload)},
+			{Name: "POLICY_MAPPING", Value: mapping.Name},
+		}
+		containers = append(containers, newInitContainer(cluster.Name, "rollover", constants.PackagedElasticsearchImage(), "/tmp/scripts/rollover", envvars))
+	} else {
+		log.V(1).Info("Skipping rollover cronjob for policymapping; hot phase not defined", "policymapping", mapping.Name)
 	}
-	schedule, err := crontabScheduleFor(policy.PollInterval)
-	if err != nil {
-		return err
+	if len(containers) == 0 {
+		log.V(1).Info("Skipping indexmanagement cronjob for policymapping; no phases not defined", "policymapping", mapping.Name)
 	}
```

Comment:
not sure this is buying us anything given we have messages above that would indicate we aren't doing hot/delete phases

Reply:
should we instead be returning `nil`?

Reply:
This only buys us an additional message that we are not creating the job at all in lieu of only constructing a portion of the job.  I can be swayed to remove it if you so desire

Reply:
I think since we are already going to be printing out that we are skipping "delete" and "hot" for when no phases are defined we can drop it... unless we do a check before this if both "hot" and "delete" are missing we would just print this message and then return early?

also, can we adjust the message? "no phases not defined"

Reply:
Just noticed... i think there is a return nil in this block which was lost during rebase.

Reply:
i think its above, there was a comment about it